### PR TITLE
fix(types): update default state typings for createStore()

### PR DESCRIPTION
**Pull request by Thienhuynh95**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A
In Stencil store main README page, it recommends to use anonymous function to return object data type for initial load. But when I use that recommendation in my project, it errors out. It turns out the types of the `createStore` function doesn't have the proper type as written in the docs. I need to patch stencil/store package for a long time in my project just to update types for `store.d.ts` to fix this typing error.
Below is the link to the docs:
https://github.com/ionic-team/stencil-store?tab=readme-ov-file#createstoretinitialstate-t----t-shouldupdate   
And here is current types for the `createStore` function:
image
And here is the error when I tried to use it:
image

## What is the new behavior?

- Using `createStore` with anonymous function would not error out
- `defaultState` types of the `createStore` would match with what was written in the docs. 
image

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

- I've been using stencil store in my project for a long time. I forked this stencil store project, changed the types of `store.ts` and built it, installed that bundled package locally. Ran `tsc` for type checking, no errors found on my current project.
- The patch file that's working on my project
image
- The declaration version of `store.ts` after running build command:
image

## Other information

N/A

### CODE REVIEW

**Review by christian-bromann — APPROVED** [review]
This makes sense to me 👍 

Let's get another approval from the @ionic-team/stencil team before we merge.

**Review by rwaskiewicz — APPROVED** [review]
Looks good to me! Thanks!

**Pull request merged**
